Refactor gas pricing functions -- Addresses Issue #1264

dcrapis opened the pull request:
### What was wrong?
Gas pricing calculations needed refactor out of vm

Related to Issue #1264

### How was it fixed?
Moved to gas_pricing.py at higher level

gurukamath commented:
Some of the static checks are failing. You can run `tox -e static` to run the checks.

gurukamath reviewed (changes requested):
Might be a good idea to name the newly created module `blobs.py` and use that for blob related gas calculations (as well as any other blobs related code in the future).

dcrapis commented:
@gurukamath addressed comments and fixed static checks. But got some test failures, summary below. Could you suggest how to fix them or help? Thank you.

=========================== short test summary info ============================
FAILED tests/prague/test_state_transition.py::test_ethereum_tests[stEIP4844-blobtransactions - BlockchainTests/GeneralStateTests/Cancun/stEIP4844-blobtransactions/blobhashListBounds7.json::blobhashListBounds7-fork_Prague-d0g0v0]
FAILED tests/prague/test_state_transition.py::test_ethereum_tests[stEIP4844-blobtransactions - BlockchainTests/GeneralStateTests/Cancun/stEIP4844-blobtransactions/blobhashListBounds8.json::blobhashListBounds8-fork_Prague-d0g0v0]
FAILED tests/prague/test_state_transition.py::test_ethereum_tests[stEIP4844-blobtransactions - BlockchainTests/GeneralStateTests/Cancun/stEIP4844-blobtransactions/blobhashListBounds9.json::blobhashListBounds9-fork_Prague-d0g0v0]
FAILED tests/prague/test_evm_tools.py::test_ethereum_tests_evm_tools[stEIP4844-blobtransactions - GeneralStateTests/Cancun/stEIP4844-blobtransactions/blobhashListBounds7.json::blobhashListBounds7-fork_[Cancun-Prague]-d0g0v0 - 0]
FAILED tests/prague/test_evm_tools.py::test_ethereum_tests_evm_tools[stEIP4844-blobtransactions - GeneralStateTests/Cancun/stEIP4844-blobtransactions/blobhashListBounds8.json::blobhashListBounds8-fork_[Cancun-Prague]-d0g0v0 - 0]
FAILED tests/prague/test_evm_tools.py::test_ethereum_tests_evm_tools[stEIP4844-blobtransactions - GeneralStateTests/Cancun/stEIP4844-blobtransactions/blobhashListBounds9.json::blobhashListBounds9-fork_[Cancun-Prague]-d0g0v0 - 0]
=== 6 failed, 294420 passed, 34 skipped, 1301 xfailed in 4700.02s (1:18:20) ====

SamWilsn commented:
Hm, looks like because the pypy tests finish first, and have backtraces turned off, you don't get a lot of useful information from CI. I'll open an issue for that.

I'll run the CPython version of the tests again to get more useful information.
